### PR TITLE
Add the Omnitool, drone adjustments

### PR DESCRIPTION
**Pull request by HoofedEar**
## About the PR 

* Add the Omnitool
    * A powerful, single slot tool that houses a wrench, a drill, a crowbar, snipping tool, and multitool all in one!
    * This tool will be exclusive to the drone. Far too powerful for mortal beings.
    * Tool speed set to 1.2x instead of the 1.5x from the Power drill and Jaws of Life.
* Change the drone's set of tools
    * Power drill and Jaws of Life -> T-Ray Scanner, Omnitool, and a satchel
    * Drone will still have access to the special welding tool it already has
* Reduce the number of free hands a drone has from 3 -> 2.

Since the Drone 1984 PR, drones can only pick up certain objects, so the satchel is just a standard satchel. It is dropped on death when the drone is gibbed.

**Screenshots**

Peek 2022-03-15 17-02

**Changelog**

:cl:
- add: Drones now have access to the Omnitool, a T-Ray Scanner, and a satchel!
- tweak: Drones have 1 less available hands.

### CODE REVIEW

**Comment by Willhelm53** [comment]
Not to scope creep this one, but the drone's number of hands could probably be reduced if this is merged.

**Comment by HoofedEar** [comment]
> Not to scope creep this one, but the drone's number of hands could probably be reduced if this is merged.

I would be super cool with this change tbh, I actually tried doing this myself but couldn’t find where the hands were defined :L

**Comment by HoofedEar** [comment]
Adjusted the PR to change drone hand size and slightly reduce their movement speed. This is meant to gear drones towards a more balanced state as a ghost role

**Comment by HoofedEar** [comment]
Merge [#7009] first before this one, as I will have to fix the conflicts. But yeah I added the bag so far, and gave them back their hands, but I will also work on all the items being dropped from the bag when they die

File: Resources/Prototypes/Entities/Mobs/Player/silicon.yml [hunk 1]
```diff
@@ -93,8 +94,8 @@
      2. You may not harm any being, regardless of intent or circumstance.
      3. Your goals are to build, maintain, repair, improve, and power to the best of your abilities, You must never actively work against these goals.
   - type: MovementSpeedModifier
-    baseWalkSpeed : 5
-    baseSprintSpeed : 5
+    baseWalkSpeed : 4
```
Comment (Elijahrane) [inline Resources/Prototypes/Entities/Mobs/Player/silicon.yml:97]: I don't think this change is needed anymore now that they slow down on hit and are stunnable

**Comment by HoofedEar** [comment]
Alright it's done, ready for review :)

**Comment by HoofedEar** [comment]
STOP THE COUNT. prying isn't working on the omnitool, one moment.

**Comment by HoofedEar** [comment]
Ok now review pls :)

File: Content.Server/Drone/DroneSystem.cs [hunk 2]
```diff
@@ -69,9 +70,16 @@ private void OnMobStateChanged(EntityUid uid, DroneComponent drone, MobStateChan
...
                 var body = Comp<SharedBodyComponent>(uid); //There's no way something can have a mobstate but not a body...
 
-                foreach (var item in drone.ToolUids)
+                foreach (var item in drone.ToolUids.Select((value, i) => ( value, i )))
                 {
-                    EntityManager.DeleteEntity(item);
+                    if (drone.Tools[item.i].PrototypeId == "ClothingBackpackSatchelDrone")
+                    {
+                        RemComp<UnremoveableComponent>(item.value);
+                    }
```
Comment (mirrorcult) [inline Content.Server/Drone/DroneSystem.cs:78]: absolutely do not hardcode this. make a tag or component

File: Content.Server/Drone/DroneSystem.cs [hunk 3]
```diff
@@ -93,12 +101,13 @@ private void OnMindAdded(EntityUid uid, DroneComponent drone, MindAddedMessage a
...
                    foreach (var entry in drone.Tools)
-                    {
-                        var item = EntityManager.SpawnEntity(entry.PrototypeId, spawnCoord);
-                        AddComp<UnremoveableComponent>(item);
-                        hands.PutInHand(item);
-                        drone.ToolUids.Add(item);
-                    }
+                   {
+                       var item = EntityManager.SpawnEntity(entry.PrototypeId, spawnCoord);
+                       if (item.ToString() != "ClothingBackpackSatchelDrone")
```
Comment (mirrorcult) [inline Content.Server/Drone/DroneSystem.cs:106]: same here with hardcoding, also just indent the if statement block normally

File: Resources/Prototypes/Body/Presets/drone.yml [hunk 4]
```diff
@@ -5,6 +5,6 @@
     hand 1: LeftHandDrone
     hand 2: LeftHandDrone
     hand 3: LeftHandDrone
-    hand 4: RightHandDrone
```
Comment (mirrorcult) [inline Resources/Prototypes/Body/Presets/drone.yml:8]: why?

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Server/Drone/DroneSystem.cs (hunk 3 above)
Comment (HoofedEar) [reply Content.Server/Drone/DroneSystem.cs:106]: ah yeah this was a holdover from another attempt to get the bag to drop. I'll remove this bit

File: Content.Server/Drone/DroneSystem.cs (hunk 2 above)
Comment (HoofedEar) [reply Content.Server/Drone/DroneSystem.cs:78]: got it, i'll change this to a tag

File: Resources/Prototypes/Body/Presets/drone.yml (hunk 4 above)
Comment (HoofedEar) [reply Resources/Prototypes/Body/Presets/drone.yml:8]: When you add items to a drone, it fills up the left hands, then the right. But this causes the first hand to have a tool, middle hand be empty, then the other 3 hands have tools. It's very weird. So switching a hand from right to left fixes this and looks better

File: Content.Server/Drone/DroneSystem.cs (hunk 2 above)
Comment (HoofedEar) [reply Content.Server/Drone/DroneSystem.cs]: complete, checks for tag instead

File: Content.Server/Drone/DroneSystem.cs (hunk 3 above)
Comment (HoofedEar) [reply Content.Server/Drone/DroneSystem.cs]: removed this

File: Resources/Prototypes/Entities/Mobs/Player/silicon.yml (hunk 1 above)
Comment (HoofedEar) [reply Resources/Prototypes/Entities/Mobs/Player/silicon.yml]: Reverted

File: Resources/Prototypes/Body/Presets/drone.yml (hunk 4 above)
Comment (Elijahrane) [reply Resources/Prototypes/Body/Presets/drone.yml:8]: He's right

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**